lib/mhv_ac/registration_form.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+require 'common/models/form'
+
+module MHVAC
+  class RegistrationForm < Common::Form
+    attribute :icn, String
```

Comment:
@patrickvinograd Here's some general information that may be helpful as you think about this.  You're right that Vets.gov is primarily Veteran-facing, but that also anyone can use it (e.g. us, active duty military, etc).  What the "anyone" else can actually do is limited currently to what is in MVI as the correlation hub for all things.  The current "gate" for access to any services via Vets.gov is "if you're not in MVI, you get nothing that requires a correlation ID, which is everything".  It is therefore possible that an active duty service member is in MVI because they have some sort of correlated use in another system e.g. MHV or eBenefits.  That's ok. Basically if you match, you match.

We have the ability to _specifically_ gate app use by are you or are you not a _veteran_ although it's not in production, nor has it actually been requested as a gate so we're not going to willy nilly throw it in.  If we _did_ throw it in, the login process would probably look like this: login to ID.me, everything is good and SAML is decrypted, check MVI for correlation IDs, use EDIPI to check for veteran status.  if not a veteran but is in mvi, restrict access to things they may actually have valid access to.  That sounds crappy and not like something we should do.

Also, it doesn't change the outcome much at all, because in order to check if someone is a veteran, we need to get the EDIPI from MVI.  If a user isn't in MVI, we will never be able to check if they're a veteran, which is effectively the same as saying if they're not in MVI they're not a veteran...which puts us in the same place as before (your flow chart would be a circle).  

cc @ayaleloehr and @robbiethegeek-usds in case further info is warranted